docs/4.1/architecture/proxy.md

```diff
@@ -0,0 +1,96 @@
+## The Proxy Service
+
+[TOC]
+
+The proxy is a stateless service which performs three main functions in a
+Teleport cluster:
+
+1. It serves as an authentication gateway. It asks for credentials from
+   connecting clients and forwards them to the Auth server via [Auth
+   API](./auth/#auth-api).
+
+2. It looks up the IP address for a requested Node and then proxies a connection
+   from client to Node.
+
+3. It serves a Web UI which is used by cluster users to sign up and configure
+   their accounts, explore nodes in a cluster, log into remote nodes, join
+   existing SSH sessions or replay recorded sessions.
+
+## Connecting to a Node
+
+### Web to SSH Proxy
+
+In this mode, Teleport Proxy implements WSS - secure web sockets - to proxy a
+client SSH connection:
+
+![Teleport Proxy Web](../img/proxy-web.svg)
+
+1. User logs in to Web UI using username, password and 2nd factor token.
```

Comment:
2-factor tokens aren't used/supported when logging in via an SSO provider - a note about this might be worthwhile.